src/TurbulenceConvection/closures/entr_detr.jl

```diff
@@ -155,48 +46,92 @@ function compute_entr_detr!(
-                εδ_model_vars = (;
-                    q_cond_up = q_cond_up, # updraft condensate (liquid water + ice)
-                    q_cond_en = q_cond_en, # environment condensate (liquid water + ice)
-                    w_up = w_up_c[k], # updraft vertical velocity
-                    w_en = w_en_c[k], # environment vertical velocity
-                    b_up = aux_up[i].buoy[k], # updraft buoyancy
-                    b_en = aux_en.buoy[k], # environment buoyancy
-                    dMdz = ∇m_entr_detr[k], # updraft momentum divergence
-                    M = m_entr_detr[k], # updraft momentum
-                    a_up = aux_up[i].area[k], # updraft area fraction
-                    a_en = aux_en.area[k], # environment area fraction
-                    RH_up = aux_up[i].RH[k], # updraft relative humidity
-                    RH_en = aux_en.RH[k], # environment relative humidity
-                    max_area = max_area, # maximum updraft area
-                    H_up = plume_scale_height[i], # plume scale height
-                    tke_en = aux_en.tke[k], # environment tke
-                )
 
-                # Compute fractional and turbulent entrainment/detrainment
-                er = entr_detr(εδ_closure, εδ_model_vars)
-                aux_up[i].entr_sc[k] = er.ε_dyn
-                aux_up[i].detr_sc[k] = er.δ_dyn
-                aux_up[i].frac_turb_entr[k] = er.ε_turb
+                # Get parameter values
+                # TODO - refactor this after we get rid of namelist stuff
+                γ_lim = εδ_params(εδ_closure).γ_lim
+                β_lim = εδ_params(εδ_closure).β_lim
+                μ_0 = εδ_params(εδ_closure).μ_0
+                β = εδ_params(εδ_closure).β
+                χ = εδ_params(εδ_closure).χ
+                c_ε = εδ_params(εδ_closure).c_ε
+                c_γ = εδ_params(εδ_closure).c_γ
+                c_λ = εδ_params(εδ_closure).c_λ
+                q_cond_up = TD.condensate(
+                    TD.PhasePartition(
+                        aux_up[i].q_tot[k],
+                        aux_up[i].q_liq[k],
+                        aux_up[i].q_ice[k],
+                    ),
+                )
+                q_cond_en = TD.condensate(
+                    TD.PhasePartition(
+                        aux_en.q_tot[k],
+                        aux_en.q_liq[k],
+                        aux_en.q_ice[k],
+                    ),
+                )
+                c_δ = if !TD.has_condensate(q_cond_up + q_cond_en)
+                    FT(0)
+                else
+                    εδ_closure.params.c_δ
+                end
+
+                # Define helper variables
+                w_up = w_up_c[k] # updraft vertical velocity
+                w_en = w_en_c[k] # environment vertical velocity
+                b_up = aux_up[i].buoy[k] # updraft buoyancy
+                b_en = aux_en.buoy[k] # environment buoyancy
+                dMdz = ∇m_entr_detr[k] # updraft momentum divergence
```

Comment:
I don't think we're using `dMdz` anymore

Reply:
Nice! Two more lines to delete!